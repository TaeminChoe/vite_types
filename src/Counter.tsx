import { useState } from "react";

const Counter = (): JSX.Element => {
  const [count, setCount] = useState<number>(0);

  const handleIncrement = () => {
    setCount(count + 1);
  };

  const handleDecrement = () => {
    setCount(count - 1);
  };

  return (
    <div>
      <h2>COUNTER</h2>
      <p>{count}</p>
      <button type="button" onClick={handleIncrement}>
        +
      </button>
      <button type="button" onClick={handleDecrement}>
        -
      </button>
    </div>
  );
};

export default Counter;
