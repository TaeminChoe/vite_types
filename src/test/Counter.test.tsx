import { fireEvent, render } from "@testing-library/react";
import Counter from "@/Counter";

describe("<Counter /> test", () => {
  it("matches snapshot", () => {
    const utils = renderCounter();
    expect(utils.container).toMatchSnapshot();
  });

  it("screen test", () => {
    const counter = renderCounter();

    const h2 = counter.container.querySelector("h2");
    h2 && expect(h2.innerHTML).toBe("COUNTER");
  });

  it("increment button test", () => {
    const counter = renderCounter();

    const count = counter.container.querySelector("p");
    const incrementButton = counter.getByText("+");

    fireEvent.click(incrementButton);
    count && expect(count.innerHTML).toBe("1");
  });

  it("decrement button test", () => {
    const counter = renderCounter();

    const count = counter.container.querySelector("p");
    const decrementButton = counter.getByText("-");

    fireEvent.click(decrementButton);
    count && expect(count.innerHTML).toBe("-1");
  });
});

const renderCounter = () => {
  return render(<Counter />);
};
