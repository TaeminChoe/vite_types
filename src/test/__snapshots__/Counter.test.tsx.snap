// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> test matches snapshot 1`] = `
<div>
  <div>
    <h2>
      COUNTER
    </h2>
    <p>
      0
    </p>
    <button
      type="button"
    >
      +
    </button>
    <button
      type="button"
    >
      -
    </button>
  </div>
</div>
`;
